File: Conexion base de datos/JAVA/MySql.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class MySQLSimpleConnection {
    public static void main(String[] args) {
        // Parámetros de conexión
        String url = "jdbc:mysql://localhost:3306/";
        String user = "root";
        String password = "123456";
        
        Connection conn = null;
        
        try {
            // 1. Registrar el driver (no necesario desde JDBC 4.0+)
            Class.forName("com.mysql.cj.jdbc.Driver");
            
            // 2. Establecer conexión
            conn = DriverManager.getConnection(url, user, password);
            
            // 3. Crear Statement (equivalente al cursor en Python)
            // (No se usa en este ejemplo ya que es solo para conexión)
            
            System.out.println("✅ Conexión exitosa a MySQL");
            
        } catch (ClassNotFoundException e) {
            System.err.println("Error: Driver JDBC no encontrado");
            e.printStackTrace();
        } catch (SQLException e) {
            System.err.println("❌ Error de conexión SQL: " + e.getMessage());
        } finally {
            // 4. Cerrar conexión
            if (conn != null) {
                try {
                    conn.close();
                } catch (SQLException e) {
                    System.err.println("Error al cerrar conexión: " + e.getMessage());
                }
            }
        }
    }
}

File: Conexion base de datos/PHP/MySql.php
<?php
$host = "localhost";
$user = "root";
$password = "123456";

try {
    // Crear conexión
    $conn = new mysqli($host, $user, $password);
    
    // Verificar conexión
    if ($conn->connect_error) {
        throw new Exception("Error de conexión: " . $conn->connect_error);
    }
    
    echo "✅ Conexión exitosa a MySQL";
    
    // Crear cursor (equivalente al cursor de Python)
    $cursor = $conn->query("SELECT 1");
    
    // Ejemplo de consulta
    $result = $conn->query("SELECT VERSION()");
    $row = $result->fetch_row();
    echo "Versión de MySQL: " . $row[0];
    
} catch (Exception $e) {
    echo "❌ Error: " . $e->getMessage();
} finally {
    // Cerrar conexión
    if (isset($conn)) {
        $conn->close();
    }
}
?>